spec/prog/postgres/postgres_server_nexus_spec.rb

```diff
@@ -1052,8 +1277,10 @@
     end
 
     it "returns true if the resource is upgrading" do
-      expect(postgres_server.resource).to receive(:target_version).and_return("17")
-      expect(postgres_server.resource).to receive(:upgrade_candidate_server).and_return(postgres_server)
+      # Set target_version different from current version to simulate an upgrade in progress
+      new_version = (postgres_server.version == "17") ? "18" : "17"
```

Comment:
We should know at this point what the version is, and hard code the version. If we don't want to do that:


```suggestion
      new_version = postgres_server.version.succ
```